src/wallet/wallet.cpp

```diff
@@ -2407,10 +2412,11 @@ CWallet::Balance CWallet::GetBalance(const int min_depth, bool avoid_reuse) cons
     {
         auto locked_chain = chain().lock();
         LOCK(cs_wallet);
+        std::set<uint256> trusted_parents;
         for (const auto& entry : mapWallet)
         {
             const CWalletTx& wtx = entry.second;
-            const bool is_trusted{wtx.IsTrusted(*locked_chain)};
+            const bool is_trusted{wtx.IsTrusted(*locked_chain, trusted_parents)};
```

Comment:
nit: I would prefer to keep using the old interface here and in 2488, 3784 until the `trusted_parents` are actually used/shared.

Reply:
What do you mean by "actually used/shared"?

Reply:
I mean for now, you could leave this line unchanged since `trusted_parents` is not used outside of `IsTrusted`. I know you eventually want to share the parents for optimization but I would not change this line until that is the case.

Reply:
I'm not sure what the "eventually" or "actually" part is... this is what is done by the code already. Note that trusted_parents lives outside the for loop so is shared across calls.

And it's less so optimization and more so anti-DoS.

Reply:
I see, I misunderstood at which level the caching is happening, thanks!